lms/djangoapps/certificates/badge_handler.py

```diff
@@ -0,0 +1,68 @@
+from PIL import Image
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.template.defaultfilters import slugify
+from io import BytesIO
+from requests import get, post
+from certificates.models import BadgeTracker
+from xmodule.contentstore.content import StaticContent
+from xmodule.modulestore.django import modulestore
+from xmodule.contentstore.django import contentstore
+
+
+class BadgeHandler(object):
+    # Global caching dict
+    badges = {}
+
+    def __init__(self, course_key):
+        self.course_key = course_key
+        # Badgr can't handle underscores at this time. Slugify, then swap out underscores with hyphens.
+        self.course_slug = slugify(unicode(course_key)).replace("_", "-")
+        self.issuer = settings.BADGR_ISSUER_SLUG
+        self.token = settings.BADGR_API_TOKEN
+        self.base_url = "{}/v1/issuer/issuers/{}".format(settings.BADGR_BASE_URL, self.issuer)
+        self.badge_create_url = "{}/badges".format(self.base_url)
+        self.badge_url = "{}/{}".format(self.badge_create_url, self.course_slug)
+        self.assertion_url = "{}/assertions".format(self.badge_url)
+        assert self.token
+
+    def get_headers(self):
+        return {'Authorization': 'Token {}'.format(self.token)}
+
+    def verify_badge(self):
```

Comment:
@Kelketek nit: I would expect `verify_*` to check some condition and return True/False. Looks like this method ensures the badge is created, so it might make sense to give him new name to reflect that, i.e. `ensure_badge_created`